sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -266,141 +264,147 @@ public virtual async Task<Response<UploadLogsResult>> UploadAsync<T>(string rule
+            foreach (var task in runningTasks)
+            {
+                // Check if an exception to log
+                if (task.CurrentTask.Exception != null)
                 {
-                    // Go through errors from each task and if we have an error, add error from response into errors list which represents the errors from all the batches
-                    if (task.Result.Error != null)
-                        errors.Add(task.Result.Error);
+                    AddException(
+                        exceptions,
+                        task.CurrentTask.Exception);
+                    logsFailed += task.LogsCount;
+                }
+                // Check status code
+                else
+                {
+                    if (task.CurrentTask.Result.Status != 204)
+                    {
+                        exceptions = AddException(
+                                    exceptions,
+                                    new RequestFailedException(task.CurrentTask.Result));
+                    }
+                    logsFailed += task.LogsCount;
                 }
             }
-            catch (Exception ex)
-            {
-                scope.Failed(ex);
-                throw;
-            }
+            ThrowException(logs, scope, exceptions, logsFailed);
 
-            // Calculate the status using the helper method Status
-            UploadLogsResult finalResult = new UploadLogsResult(errors, GetStatus(logs, errors));
-            return Response.FromValue(finalResult, response);
+            // If no exceptions return response
+            return runningTasks.Select(_ => _.CurrentTask).Last().Result; //204 - response of last batch with header
         }
 
-        private async Task<BatchUpload> CommitBatchListSyncOrAsync<T>(BatchedLogs<T> batch, string ruleId, string streamName, bool async, CancellationToken cancellationToken)
+        private async Task<Response> UploadBatchListSyncOrAsync<T>(BatchedLogs<T> batch, string ruleId, string streamName, bool async, CancellationToken cancellationToken)
         {
-            UploadLogsError error = null;
-            RequestContext requestContext = GenerateRequestContext(cancellationToken);
             Response response = null;
 
-            using HttpMessage message = CreateUploadRequest(ruleId, streamName, batch.LogsData, "gzip", requestContext);
+            using HttpMessage message = CreateUploadRequest(ruleId, streamName, batch.LogsData, Compression, null);
 
             if (async)
             {
-                response = await _pipeline.ProcessMessageAsync(message, requestContext, cancellationToken).ConfigureAwait(false);
+                response = await _pipeline.ProcessMessageAsync(message, null, cancellationToken).ConfigureAwait(false);
```

Comment:
We can use pipeline.Send

Reply:
I'm not sure of the difference between ProcessMessage and Send - the protocol methods generated ProcessMessage so I kept it the same. Maybe it should be updated?

Reply:
ProcessMessage is for use with RequestContext. Since our convenience methods don't expose RequestContext parameter, we can simplify to use Send.